great_expectations/rule_based_profiler/domain_builder/types/domain.py

```diff
@@ -1,16 +1,37 @@
 import json
+from dataclasses import asdict, dataclass
+from enum import Enum
 from typing import Any, Dict, Optional, Union
 
 from great_expectations.core import IDDict
 from great_expectations.core.util import convert_to_json_serializable
 from great_expectations.execution_engine.execution_engine import MetricDomainTypes
-from great_expectations.rule_based_profiler.domain_builder.inferred_semantic_domain_type import (
-    SemanticDomainTypes,
-)
+from great_expectations.types import SerializableDictDot
 from great_expectations.types.base import SerializableDotDict
 from great_expectations.util import filter_properties_dict
 
 
+class SemanticDomainTypes(Enum):
+    NUMERIC = "numeric"
+    TEXT = "text"
+    LOGIC = "logic"
+    DATETIME = "datetime"
+    BINARY = "binary"
+    CURRENCY = "currency"
+    VALUE_SET = "value_set"
+    MISCELLANEOUS = "miscellaneous"
+    UNKNOWN = "unknown"
```

Comment:
I think we agreed this a property of `InferredSemanticTypeDomainBuilder`, so that subclasses of that builder and update/modify it as needed.

Reply:
@jcampbell Happy to move it to whatever the best place is.  For subclassing, maybe it should be in a separate module by itself?  Not sure; let's discuss -- and easy enough to make the change as needed.

Reply:
On reflection -- I think it's okay to leave the *definition* here as long as it's included as a property in the DomainBuilder and *we* use it *via* the SemanticTypeDomainBuilder -- can you confirm that that is indeed the case?

Reply:
@jcampbell To be very precise: A field having the type of `SemanticDomainTypes` (which is an `Enum`) is a property of `InferredSemanticDomainType` *only*; furthermore, only `SimpleSemanticTypeColumnDomainBuilder` (and the various custom implementations) are able to infer semantic types from physical columns (and hence set this property).

Reply:
@jcampbell Being convinced that this is a bug, I fixed it as proposed above.  If this is in error, please let me know.  Thanks.